### PR TITLE
Media previews

### PR DESCRIPTION
**Pull request by simplesthing**
- Fixes digitalworkersguild/banchan#161
- Resolves image upload part of digitalworkersguild/banchan#160

### CODE REVIEW

**Comment by simplesthing** [comment]
Not sure if the test is correct ,if it looks good I can add other file formats.

**Comment by zkat** [comment]
I've fixed the bug that I think was causing the test failure you're seeing. Please rebase onto main and give it another shot :)

File: lib/banchan/uploads/uploads.ex [hunk 1]
```diff
@@ -56,15 +60,15 @@ defmodule Banchan.Uploads do
...
+    Repo.one!(from(u in Upload, where: u.bucket == ^bucket and u.key == ^key))
   end
 
   def save_file!(%User{} = user, src, type, file_name, bucket \\ get_bucket()) do
     key = gen_key()
     size = File.stat!(src).size
 
     {width, height} =
-      if image?(type) || video?(type) do
+      if image?(type) do
```
[the remark this answers is not in the document]
Comment (simplesthing) [reply lib/banchan/uploads/uploads.ex:71]: It seems to be failing for me locally since it was added, does it work on other machines?

File: test/banchan/commissions_test.exs [hunk 2]
```diff
@@ -149,6 +149,20 @@ defmodule Banchan.CommissionsTest do
     end
   end
 
+  describe "events" do
+    test "make_attachment" do
```
[the remark this answers is not in the document]
Comment (simplesthing) [reply test/banchan/commissions_test.exs:153]: Thank you I am still reading about tests and I wasn't sure if I caused the issue or not

File: lib/banchan/commissions/commissions.ex [hunk 3]
```diff
@@ -643,25 +643,43 @@ defmodule Banchan.Commissions do
...
+              image =
+                Uploads.save_file!(user, final_path, "image/" <> format, "thumbnail." <> format)
+
+              File.rm!(tmp_file)
+              File.rm!(final_path)
+
+              image
+
+            Uploads.video?(upload) ->
+              nil
```
[the remark this answers is not in the document]
Comment (simplesthing) [reply lib/banchan/commissions/commissions.ex:680]: There is an issue with `ffmpex`  at the moment for local dev on Mac arm M1 machines, I need to test this branch on WSL and also look for more solutions to the M1 issue. 

tldr; `ffmpex` is dependent on `rambo` which requires an M1 build, but the maintainer has not been around to merge this PR https://github.com/jayjun/rambo/pull/13

File: lib/banchan/uploads/uploads.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (simplesthing) [reply lib/banchan/uploads/uploads.ex:71]: It's possible it is failing on M1 due to the `rambo` dep mentioned below, I'll add this to my WSL tests today

File: test/banchan/commissions_test.exs [hunk 4]
```diff
@@ -149,6 +149,20 @@ defmodule Banchan.CommissionsTest do
...
+      user = user_fixture()
+      src = "test/support/fixtures/file-types/image/squid.png"
```
[the remark this answers is not in the document]
Comment (zkat) [reply test/banchan/commissions_test.exs:155]: please! :)

File: lib/banchan/commissions/commissions.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (zkat) [reply lib/banchan/commissions/commissions.ex:680]: Oof. Yeah, that's unfortunate. Hopefully it works on WSL.

File: test/banchan/commissions_test.exs (hunk 2 above)
[the remark this answers is not in the document]
Comment (zkat) [reply test/banchan/commissions_test.exs:153]: I already fixed it for you :)))

**Comment by zkat** [comment]
heya! Have you had any time to take another look at this?

**Comment by simplesthing** [comment]
I have not yet, will find time this week to wrap it up.

**Comment by simplesthing** [comment]
Closing to open a new branch for Oban update
